0.31

fabaff opened the pull request:
Hello again friend,

How are you? Having a good day? We sure are. If you aren't having a good day, this might cheer you up...

Every other weekend around here gets a little hectic leading to a big sigh of relief as we release a new version of Home Assistant to the world. And this time is no different. Our developer community has once again built us a beautiful new release with lots of new features and improvements. We hope you like it.

One last thing before we get going though, I should warn you... [@balloob](https://github.com/balloob) got a bit lazy this week and let me ([@robbiet480](https://github.com/robbiet480)) step up to the plate again to write the blog post and do the release. I guess I didn't do such a bad job in [0.27](/blog/2016/08/28/notifications-hue-fake-unification/). You'll never know what surprises I have in store. Now that i've got all that stuff out of the way, let's get started...
## Stats Update

Sadly, no big amazing stats to update you with this time, but we did recently pass 7,000 commits! This release featured submissions from 45 contributors. Hopefully with the new updater component we will be able to give you some really good stats in the 0.32 blog post.
## Hacktoberfest

![Hacktoberfest logo](https://home-assistant.io/images/blog/2016-10-hacktoberfest/hacktoberfest.png)

October means Hacktoberfest time and our community has really come through with some excellent improvements and additions. As of this writing, we have 194 merged and 41 open pull requests to the [home-assistant repository](https://github.com/home-assistant/home-assistant/labels/Hacktoberfest) and 209 merged/28 open pull requests submitted to the [home-assistant.github.io repository](https://github.com/home-assistant/home-assistant.github.io/labels/Hacktoberfest). If you want to get in on the fun check out our [Hacktoberfest](/blog/2016/10/02/hacktoberfest/) blog post or the [Hacktoberfest website](https://hacktoberfest.digitalocean.com/). You get an awesome t-shirt for free if you have 4 pull requests merged in the month of October! We even have tasks that a non-developer can easily accomplish with a tiny bit of work. Better hurry up though, only 9 days left and most of the easy tasks are gone!
## ⚠️ A greatly improved updater component (Please read this!) ⚠️

This release includes an update to our [updater](https://home-assistant.io/components/updater/) component. The responsibility of the updater component is to check if a new version is available and notify the user if this is the case.

It used to be that this component would check with [PyPi](the Python package manager) to see if a new update was available. This had a couple of problems:
1. We are unable to do a slow rollout
2. We are unable to show the user extra information (like a link to a changelog or the release date)
3. We are unable to warn users for critical security updates

So to work around these problems, we decided to start hosting the version check service ourselves. Since we had to get some infrastructure spun up anyway, we figured we would take it a step further. Which leads me to this bit of the update (the most important part):
### What you need to know (the important bit!)

Remember how I mentioned that up there in the title that there is some serious business in this release? Well, we also added some basic analytics to the updater component which get sent to the server and stored so that we get a better idea of our user base.

Each Home Assistant instance running the updater component will generate a [unique ID](https://en.wikipedia.org/wiki/Universally_unique_identifier) (based on UUIDv4) that will be used for the updater to be able to differentiate between instances. This UUID will be stored in your config directory in a file called `.uuid`.
#### Opting out

There are two ways to opt-out. The first way is by using the new `opt_out` option for the updater. This way the updater will continue to check for updates, but no information about your system will be shared with us.

``` yaml
updater:
  reporting: no
```

You can also disable the updater component entirely by removing `updater:` from your `configuration.yaml` although **we would not suggest you do this** as you would miss any critical security updates.

Finally, you can also reset your unique identifier by deleting the `.uuid` file and restarting Home Assistant.
#### Data stored on the Home Assistant update server

Here is what my production Home Assistant instance looks like from the server side:

| Name | Description | Example | Data Source |
| --- | --- | --- | --- |
| `arch` | CPU Architecture | `x86_64` | Local Instance |
| `distribution` | Linux Distribution name (only Linux) | `Ubuntu` | Local Instance |
| `docker` | True if running inside Docker | `false` | Local Instance |
| `first_seen_datetime` | First time instance ID was submitted | `2016-10-22T19:56:03.542Z` | Update Server |
| `geo_city` | GeoIP determined city | `Oakland` | Update Server |
| `geo_country_code` | GeoIP determined country code | `US` | Update Server |
| `geo_country_name` | GeoIP determined country name | `United States` | Update Server |
| `geo_latitude` | GeoIP determined latitude | `37.8047` | Update Server |
| `geo_longitude` | GeoIP determined longitude | `-122.2124` | Update Server |
| `geo_metro_code` | GeoIP determined metro code | `807` | Update Server |
| `geo_region_code` | GeoIP determined region code | `CA` | Update Server |
| `geo_region_name` | GeoIP determined region name | `California` | Update Server |
| `geo_time_zone` | GeoIP determined time zone | `America/Los_Angeles` | Update Server |
| `geo_zip_code` | GeoIP determined zip code | `94602` | Update Server |
| `last_seen_datetime` | Most recent time instance ID was submitted | `2016-10-22T19:56:03.542Z` | Update Server |
| `os_name` | Operating system name | `Darwin` | Local Instance |
| `os_version` | Operating system version | `10.12` | Local Instance |
| `python_version` | Python version | `3.5.2` | Local Instance |
| `timezone` | Timezone | `America/Los_Angeles` | Local Instance |
| `user_agent` | User agent used to submit analytics | `python-requests/2.11.1` | Local Instance |
| `uuid` | Unique identifier | `10321ee6094d4a2ebb5ed55c675d5f5e` | Local Instance |
| `version` | Home Assistant version | `0.31.0` | Local Instance |
| `virtualenv` | True if running inside virtualenv | `true` | Local Instance |

In addition to the above collected data, the server will also use your IP address to do a geographic IP address lookup to determine a general geographic area that your address is located in. To be extremely, extremely clear about this bit: **The Home Assistant updater does not: store your IP address in a database and also does not submit the location information from your `configuration.yaml`.** Our tests show that at best, we get 4 digits of accuracy on your IP address location which is a 5 mile radius of your actual IP location, assuming that it is even correct in the first place (geo IP look ups are very hit or miss). Here's what the accuracy looks like for my data above: ![Robbie's GeoIP accuracy level](https://home-assistant.io/images/blog/2016-10-flash-briefing-updater-hacktoberfest/map.png)

The server also adds two timestamps to the data: the original date your instance UUID was first seen and the timestamp of the last time we have seen your instance.

This data is held in the highest security. The update system runs in a secured Amazon Web Services account owned by me ([@robbiet480](https://github.com/robbiet480)). I personally have 5 years of experience with complex AWS deployments and have an extensive security background. I have audited the entire system and made sure to take every step to protect the data, including limiting who has access (just [@balloob](https://github.com/balloob) and myself). While not directly personally identifiable we absolutely understand some users hesistance to giving this information out. Please understand that we are only collecting this information to better understand our user base to provide better long term support and feature development then is currently possible.

We currently have no plans to publicly expose any of this information. If we did do such a thing in the future we would of course notify you in advance. It must also be stated that we will never sell or allow the use of this information for non-Home Assistant purposes.

We thank you for understanding why we are collecting this data and hope that you leave the feature enabled but fully understand if you feel uncomfortable with this.

Now, back to the fun stuff...
## Good evening. I'm Ron Burgundy and here's what happening in your world tonight.

Home Assistant got a crazy idea recently that it couldn't do enough already and wanted to challenge itself even more. I really don't understand how it came up with this kooky idea, but it now thinks that its newest hobby should be a minor career in journalism.

0.31 adds support for the brand spanking new [Alexa Flash Briefing API](https://developer.amazon.com/alexa-skills-kit/flash-briefing), allowing you to get updates from Home Assistant anytime you ask Alexa to read your flash briefing. What's the use case you ask? Well, now when I wake up in the morning and get my flash briefing, Home Assistant adds this to the end of it for me:

> Drive time with traffic is 35 minutes. There is an UberPOOL that will cost $11.52, estimated to be 2 minutes away, for a total of 37 minutes. BART is currently estimated to take 29 minutes. You should take BART, as it is estimated to be faster by 8 minutes.

Now I know how to best get to [my real job](https://runway.is) (no, Home Assistant is _not_ my real job, it does seem like it sometimes though) every morning. Obviously not the best home automation example, but I think you get the idea. I could see this being used to tell you any major events that happened in your home overnight or reading you your hyperlocal weather report. Thanks to the audio support you could even replace all of the default Alexa Flash Briefing sources with your own news feeds. Home Assistant supports both text and audio content as well as displaying data in the Alexa app. I also want to point out that unlike the existing Skill integration, the Flash Briefing API does _not_ require HTTPS (_but you should still be using HTTPS if possible_). For more information, check out the new [docs](https://home-assistant.io/components/alexa/).

![You stay classy, San Diego. (It's funny, because balloob lives in San Diego))](https://home-assistant.io/images/blog/2016-10-flash-briefing-updater-hacktoberfest/stay-classy.jpg)
## Major breaking Z-Wave changes ahead

A rather nasty Z-Wave [issue](https://github.com/home-assistant/home-assistant/pull/3759) was discovered recently by [@lukas-hetzenecker](https://github.com/lukas-hetzenecker). There was a somewhat large chance that if you had multiple of the same model Z-Wave device they may both try to use the same entity IDs. To fix the issue the internal Z-Wave index is now appended to the end of all Z-Wave entity IDs.

What this means for all you Z-Wave users is that you will need to update your configurations to reflect the change. I personally have quite a few (17) Z-Wave devices and went through the process this week. Here's what I had to do:
- Update customizations
- Update groups
- Update my `zwave.customize` section
- Update my Alexa skills that used old entity IDs
- Because I use `emulated_hue` with Alexa and `emulated_hue` uses the entity ID as a unique identifier I also had to remove all Z-Wave devices from Alexa and re-add them.

Your todo list may be a little different from mine, I just wanted to outline the steps I took to give you an idea of what you need to think about. It's not a very hard process, especially when using global find and replace in Sublime Text but did take me about 20 minutes to complete.

This is super annoying, I know, especially since we had said in [0.12](/blog/2016/01/30/insteon-lifx-twitter-and-zigbee/#backwards-incompatible-changes) that Z-Wave IDs should hopefully never change again, but we are now forced to eat those words. I will state again that Z-Wave IDs shouldn't change in the future but obviously we see how that went. To sum up on this section... sorry but it had to happen.
## All changes
- [Updater](https://home-assistant.io/components/updater/) component ([@infamy](https://github.com/infamy), [@robbiet480](https://github.com/robbiet480), [@kellerza](https://github.com/kellerza))
- Continue to setup other platforms when 1 platform config is invalid ([@kellerza](https://github.com/kellerza))
- Create [persistent notifications](https://home-assistant.io/components/persistent_notification/) when a platform contains invalid config ([@kellerza](https://github.com/kellerza))
- Logbook: Allow [filtering](https://home-assistant.io/components/logbook/) domains and entities to be shown ([@wokar](https://github.com/wokar))
- HTTP: Change `approved_ips` from string to CIDR validation ([@mweinelt](https://github.com/mweinelt))
- Persistent Notifications: Allow using [markdown](https://home-assistant.io/components/persistent_notification/) ([@justweb1](https://github.com/justweb1))
- Netatmo: Add [discovery](https://home-assistant.io/components/netatmo/) support ([@jabesq](https://github.com/jabesq))
- Netatmo Welcome camera: Add [binary sensor](https://home-assistant.io/components/binary_sensor.netatmo/) ([@jabesq](https://github.com/jabesq))
- Support added for [HaveIBeenPwned]([@joyrider3774])
- Device tracker: `known_devices.yaml` reading and writing tweaks and fixes ([@kellerza](https://github.com/kellerza))
- Fix climate platforms showing the wrong temperature unit ([@rcloran](https://github.com/rcloran))
- Lots of voluptuous love ([@fabaff](https://github.com/fabaff))
- Ensure proper attribution for weather platforms ([@fabaff](https://github.com/fabaff))
- Fix Telegram in Docker ([@jeanregisser](https://github.com/jeanregisser))
- Support recursive config inclusions in YAML ([@lwis](https://github.com/lwis))
- Camera: [Synology](https://home-assistant.io/components/camera.synology/) SS cameras now supported ([@jgriff2](https://github.com/jgriff2))
- History: Allow filtering domains and entities to be shown ([@wokar](https://github.com/wokar))
- Media Player - Squeezebox: Now able to show artist and album ([@ih8gates](https://github.com/ih8gates))
- Alexa: Flash Briefing skill support added ([@robbiet480](https://github.com/robbiet480))
- Device Tracker: Add support for Bbox Modem Router ([@HydrelioxGithub](https://github.com/HydrelioxGithub))
- Sensor: Add support for Bbox Modem Router ([@HydrelioxGithub](https://github.com/HydrelioxGithub))
- Input select: Services added to pick next and prev option ([@persandstrom](https://github.com/persandstrom))
- Sensor: [ARWN](https://home-assistant.io/components/sensor.arwn/) now supported ([@sdague](https://github.com/sdague))
- Pushbullet: Push an URL note if an url is provided inside data ([@jabesq](https://github.com/jabesq))
- Z-Wave: Allow certain devices to be not added to Home Assistant ([@lukas-hetzenecker](https://github.com/lukas-hetzenecker))
- New support for [Zoneminder](https://home-assistant.io/components/zoneminder/) added ([@Khabi](https://github.com/Khabi))
- Weather: Allow tracking severe weather alerts with [WUnderground]([@tchellomello])
- Sensor: New support added to track [min/max/mean](https://home-assistant.io/components/sensor.min_max/) ([@fabaff](https://github.com/fabaff))
- Convert EntityComponent to be async ([@pvizeli](https://github.com/pvizeli), [@balloob](https://github.com/balloob))
- Z-Wave: Add assocation service ([@turbokongen](https://github.com/turbokongen))
- Frontend - Services dev tool: persist state and tweak UI ([@justweb1](https://github.com/justweb1))
- Sensor: Support added for [scraping](https://home-assistant.io/components/sensor.scrape/) websites ([@fabaff](https://github.com/fabaff))
- Clean up of tests ([@capellini](https://github.com/capellini))
- New `fail` filter added to templates to raise on UndefinedError ([@jaharkes](https://github.com/jaharkes))
- Support added for [Emoncms history](https://home-assistant.io//components/emoncms_history/) ([@joyrider3774](https://github.com/joyrider3774))
- Support for [Apple Push Notifications Service](https://home-assistant.io/components/notify.apns/) ([@sam-io](https://github.com/sam-io))
- Thermostat: Netatmo now supported ([@gieljnssns](https://github.com/gieljnssns))
- Alarm control panel: [Concord232](https://home-assistant.io/components/alarm_control_panel.concord232/) now supported ([@JasonCarter80](https://github.com/JasonCarter80))
- Notify: [Matrix](https://home-assistant.io//components/notify.matrix/) support added ([@mweinelt](https://github.com/mweinelt))
- Device tracker - nmap: Allow specifying multiple inputs for [nmap]([@hcooper])
- Device Tracker - snmp: SNMPv3 now supported ([@T3m3z](https://github.com/T3m3z))
- Notify: [Telstra](https://home-assistant.io/components/notify.telstra/) SMS now supported ([@nvella](https://github.com/nvella))
- Camera: [Verisure](https://home-assistant.io/components/camera.verisure/) now supported ([@turbokongen](https://github.com/turbokongen))
- Support added for [Neato](https://home-assistant.io/components/switch.neato/) Connected Robot ([@jabesq](https://github.com/jabesq))
- Media player: More options for [Yamaha](https://home-assistant.io/components/media_player.yamaha/) AVR ([@ehagan](https://github.com/ehagan))
- Sensor: Support for [Pilight](https://home-assistant.io/components/sensor.pilight/) sensor ([@DavidLP](https://github.com/DavidLP))
- iOS support ([@robbiet480](https://github.com/robbiet480))
- Minor features and bug fixes by [@mtl010957](https://github.com/mtl010957), [@molobrakos](https://github.com/molobrakos), [@flyte](https://github.com/flyte), [@fabaff](https://github.com/fabaff), [@phardy](https://github.com/phardy), [@sander76](https://github.com/sander76), [@T3m3z](https://github.com/T3m3z), [@c-w](https://github.com/c-w), [@balloob](https://github.com/balloob), [@robbiet480](https://github.com/robbiet480), [@StaticCube](https://github.com/StaticCube), [@vittoriom](https://github.com/vittoriom), [@hartmms](https://github.com/hartmms), [@kirichkov](https://github.com/kirichkov), [@mezz64](https://github.com/mezz64), [@ishults](https://github.com/ishults), [@Danielhiversen](https://github.com/Danielhiversen) and [@tchellomello](https://github.com/tchellomello).
## Breaking changes
- The [HTTP](https://home-assistant.io/components/http/) component now takes a different format for authenticating IPs
- Config format has changed for [Proximity](https://home-assistant.io/components/proximity/)
- The Z-Wave entity ID change mentioned above
## If you need help...

...don't hesitate to use our [Forum](https://community.home-assistant.io/) or join us for a little [chat](https://gitter.im/home-assistant/home-assistant). The release notes have comments enabled but it's preferred if you the former communication channels. Thanks.
## Until next time

Thanks for reading all of the above, especially since this week was a pretty long post. We should be back with a new post around November 5th announcing the arrival of 0.32.

-- Robbie

balloob committed:
Version bump to 0.31.0.dev0

jabesq committed:
Add discovery support for Netatmo weather Station (#3714)

* Add discovery support for Netatmo Weather station

Only The weather information are currently supported (No battery or wifi status supported)

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

* Add unique_id for netatmo sensors to avoid duplicate

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

* Update requirements_all.txt and PEP8 fixes

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

wokar committed:
add include configuration to logbook (#3739)

kellerza committed:
Continue on invalid platforms and new setup_component unit tests (#3736)

joyrider3774 committed:
Haveibeenpwned sensor platform (#3618)

* Initial version of "haveibeenpwned" sensor component

* 2 flake8 fixes

* remove debugging error message

* Increase scan_interval as well as throttle to make sure that during initial startup of hass the request happens with 5 seconds delays and after startup with 15 minutes delays. Scan_interval is increased also to not call update as often

* update .coveragerc

* remove (ssl) verify=False

* - use dict to keep the request values with email as key
- use track_point_in_time system to make sure data updates initially at 5 seconds between each call until all sensor's email have a result in the dict.

* fix a pylint error that happend on the py35 tests

mtl010957 committed:
Fixed issue #3760, handle X10 unit numbers greater than 9. (#3763)

roidayan committed:
Fix command line cover template (#3754)

The command line cover value template is optional so we
need to check it's not none before assigning hass to it.

Fixes #3649

Signed-off-by: Roi Dayan <roi.dayan@gmail.com>

kellerza committed:
Coerce device IDs from known_devices to be slugs (#3764)

* Slugify & consider_home test fix [due to load valid PR]

* undo schema change

* Fix slugify error

fabaff committed:
Use voluptuous for Russound RNET (#3689)

* Migrate to voluptuous

* Remove wrong default

mweinelt committed:
Http: Change approved_ips from string to cidr validation (#3532) [BREAKING CHANGE]

* Change approved_ips from string to cidr validation

Relabel to trusted_networks, better reflecting its expected inputs,
everything that ipaddress.ip_networks recognizes as an ip network
is possible:
- 127.0.0.1      (single ipv4 addresses)
- 192.168.0.0/24 (ipv4 networks)
- ::1            (single ipv6 addresses)
- 2001:DB8::/48  (ipv6 networks)

* Add support for the X-Forwarded-For header

jabesq committed:
Netatmo binary sensor (#3455)

* Basic support for Netatmo welcome binary sensors

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

* Bug fixes and optimization for Netatmo devices

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

* pylint fixes

* Bug Fixing and optimization for Netatmo devices

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

* Add unique_id for cameras to avoid duplicate
And global config to disable discovery for netatmo devices

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

molobrakos committed:
slugify (#3777)

persandstrom committed:
vsure0.11.0

persandstrom committed:
Merge pull request #3782 from persandstrom/vsure0.11.0

vsure0.11.0

pvizeli committed:
Hotfix device name with autodiscovery (#3791)

flyte committed:
Make 'pin' optional for zigbee device config. (#3799)

fabaff committed:
No longer use old name (#3792)

fabaff committed:
Update docstrings (sensor.pi_hole, sensor.haveibeenpwned) (#3793)

* Fix docstrings

* Update docstrings

sander76 committed:
Changing import as powerview api did change. (#3780)

robbiet480 committed:
Wrap found target in list (#3809)

* Wrap found target in list

* Fix test_messages_to_targets_route

phardy committed:
Stop GTFS component from overwriting friendly_name (#3798)

* Prevent update from clobbering configured name.

* linted

* Fixing awkward line breaking.

T3m3z committed:
Fix for html5 notification tag problem. Fixes #3774 (#3790)

fabaff committed:
Upgrade pyowm to 2.5.0 (#3806)

fabaff committed:
Upgrade dnspython3 to 1.15.0 (#3804)

fabaff committed:
Upgrade cherrypy to 8.1.2 (#3805)

fabaff committed:
Upgrade sqlalchemy to 1.1.1 (#3796)

c-w committed:
Cache condition in helpers.Script (#3797)

* Cache condition in helpers.Script

The caching is a simple in-memory, per-instance dictionary.
We use __str__ to format cache keys.

This naive implementation has some disadvantages (e.g., we won't be able
to cache two conditions that contain references to
distinct-but-equivalent object instances and we don't have any control
over the size of the condition cache), but for most simple use-cases the
approach should be good enough.

Resolves #3629

* Fix docstring style

lukas-hetzenecker committed:
Update python-lirc to 1.2.3 (#3784)

* Upgrade to latest python-lirc 1.2.3

* update requirements_all.txt

ProtoThis committed:
Added additional checks which hides functions which are not supported by Nest (#3751)

* Added additional checks which hides functions which are not support (like fans / humidity / cooling)

* Fixed pylint and flake8 errors (not test file available)

* Fixed pydocstyle error

* Refactored Code and Comments as described in pull-request

* Added additional comment and requesting retest

* Upgraded to python-nest 2.11 which contains previously hidden functions

rcloran committed:
Separate climate platform and presentation units (#3755)

* Separate platform and presentation units in climate

* Fix unit tests

Maybe

* Fix unit tests some more

Maybe

* Rename _platform_unit_of_measurement to temperature_unit

* Fix tests for renamed attribute

fabaff committed:
Migrate to voluptuous (#3293) [BREAKING CHANGE]

fabaff committed:
Use voluptuous for FitBit (#3686)

* Migrate to voluptuous

* Fix default

fabaff committed:
Add/adjust attribution of sensor platform (#3719)

* Add/adjust attribution

* Fix typo

fabaff committed:
Use voluptuous for Heatmiser (#3732)

fabaff committed:
Migrate to voluptuous (#3737)

balloob committed:
Update recorder callback (#3812)

fabaff committed:
Use voluptuous for eQ-3 thermostat (#3729)

* Migrate to voluptuous

* Fix requirement and typo

fabaff committed:
Migrate to voluptuous (#3817)

robbiet480 committed:
Notify: Only attach target if in call data (#3831)

* Only pass through the target if it has a value

* Target will no longer be none

fabaff committed:
Fix slack targets (#3826)

Nixon506E committed:
Restore Optional Target Config Attribute (notify.pushover) (#3769)

* Restore Optional Target Config Attribute

* Fix Tabs

* Change indents to spaces

* Make a target fix

* Change to simpler not syntax

jeanregisser committed:
Add again certifi to Docker image (#3813)

Latest versions of certifi (>= 2016.8.31) don't seem to break anything
anymore.
See home-assistant/home-assistant#2554

lwis committed:
Support recursive config inclusions (#3783)

jgriff2 committed:
Add synology ss cameras (#3626)

* Add files via upload

* Update .coveragerc

* test

* Update synology camera

* Use voluptuous for synology

* Use voluptuous for synology

* Use voluptuous for synology

* Use voluptuous for synology

* Conform synology to flake8

* Added Whitelist to synology

* Sync to dev branch

* Added helper function to synology

vittoriom committed:
Fixes an issue where Chromecast audio groups were not properly discovered (#3630)

* Fixes an issue where Chromecast audio groups were not properly discovered

* Forgot to commit the main fix

* Removes unused variable

* Doesn't use a protected API anymore

* PR remarks

* Fixes tests, adds comment

* Restores line as it was in the original commit, rephrases comment

* Should fix lint issues

* Trailing whitespace

* Some more lint

wokar committed:
Added include and exclude functionality to history component (#3674)

* added include and exclude functionality to history component

* fixed summary lines in test method doc.

* cleanup of query filter creation

* o improved config validation
o move move IGNORE_DOMAINS to Filter.apply()
o removed config from Last5StatesView
o Filters instance is now created on setup
o config values are processed in setup and set to the Filters instance
o function _set_filters_in_query() moved to Filters class and renamed to apply()

* added unittests for more include/exclude filter combinations

* make pylint happy

ih8gates committed:
Proper title, added album and artist for Squeezebox (#3735)

* Proper title, added album and artist

Title had previously concatenated artist - title.

* Made changes suggested by @balloobbot

kellerza committed:
Add persistent notifications to bootstrap (#3738)

* Add persistent notifications to bootstrap

* Rebase, Fix test

mezz64 committed:
Update zwave lights to increase update delay timer as needed (#3741)

Fix permissions

robbiet480 committed:
Add Alexa Flash Briefing Skill API support (#3745)

* Add Alexa Flash Briefing Skill API support

* Set default value for text to empty string as per API docs

* Clean up existing Alexa tests

* Update configuration parsing and validation

* Add tests for the Flash Briefing API

* Update test_alexa.py

fabaff committed:
Use async_render_* and fix config validation (#3847)

jabesq committed:
Hotfix for Netatmo discovery (#3837)

This should definetly fix #2601

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

tchellomello committed:
Fixed Fitbit resting heart rate attribute (#3835)

* Fixed Fitbit restingHeartRate field to grab the correct field from dictionary

In [31]: r['activities-heart'][-1].get('value')
Out[31]:
{'customHeartRateZones': [],
 'heartRateZones': [{'caloriesOut': 126.18348,
   'max': 94,
   'min': 30,
   'minutes': 67,
   'name': 'Out of Range'},
  {'caloriesOut': 27.21339,
   'max': 131,
   'min': 94,
   'minutes': 5,
   'name': 'Fat Burn'},
  {'caloriesOut': 0, 'max': 159, 'min': 131, 'minutes': 0, 'name': 'Cardio'},
  {'caloriesOut': 0, 'max': 220, 'min': 159, 'minutes': 0, 'name': 'Peak'}],
 'restingHeartRate': 69}

In [32]: r['activities-heart'][-1].get('value').get('restingHeartRate')
Out[32]: 69

* Renamed sensor to Resting Heart Rate to match it

* Fixed lint style

jaharkes committed:
Fail when rendering undefined objects in Jinja2 templates

By not successfully rendering unknown objects to an empty string
the caller provided error_value actually gets used.

This allows, for instance, the MQTT sensor to retain its state when an
unexpected or unwanted message (#2733/#3834) is received.

persandstrom committed:
vsure 0.11.1

persandstrom committed:
Merge pull request #3857 from persandstrom/vsure0.11.1

vsure 0.11.1

HydrelioxGitHub committed:
Add Bbox Modem Routeur for device tracker (#3848)

lukas-hetzenecker committed:
Include index and instance in object_id of zwave devices (#3759)

* Include index and instance in object_id of zwave devices

* Add the instance id if there is more than one instance for the value

jaharkes committed:
Bump pychromecast dependency to 0.7.6 (#3864)

persandstrom committed:
select next and previous of input select (#3839)

turbokongen committed:
Use only node id to identify node in set_config_parameter (#3801)

balloob committed:
Update frontend

sdague committed:
add arwn sensor platform (#3846)

This adds a sensor component that builds sensors based on the arwn
project (https://github.com/sdague/arwn). This uses a 433mhz receiver
to collect weather data and publish it over mqtt in a well defined
schema, which home-assistant can display as sensors.

sdague committed:
Bump proliphix library to 0.4.0 (#3855)

There was a bug in setback setting which is now fixed in 0.4.0. This
ensures that any users have working setback code.

turbokongen committed:
Use pass instead of return None (#3856)

ishults committed:
Allow any positive integer for Z-Wave polling intensity (#3859)

jabesq committed:
Pushbullet push an url note if an url is provided inside data (#3758)

Commit:
update flux led library

Danielhiversen committed:
Merge pull request #3868 from home-assistant/flux_led_lib

update flux led library

jaharkes committed:
Use a filter to fail rendering undefined variables

Instead of globally using StrictUndefined, introduce a filter that
will trigger a render failure on undefined variables.

Use as `{{ value_json.someval | is_defined }}`.

lukas-hetzenecker committed:
Add ignore option to zwave customize configuration (#3865)

Khabi committed:
Zoneminder component (#3795)

* Initial Zoneminder commit

* Fixing bug when ZM sets its function to 'None'

* Adding zoneminder to coverage

* Quick Doc fix

* Update zoneminder.py

Doc Fix

* making the url base optional

fabaff committed:
Upgrade slacker to 0.9.28 (#3877)

fabaff committed:
Upgrade uber_rides to 0.2.7 (#3876)

hartmms committed:
Catch MQTT encoding errors (#3749)

* added error handling to mqtt message receive if payload is not utf-8 unicode
added mqtt test for above code as well

* change permission back to 644

* attempting to test new code

* changed exception to AttributeError
fixed test for above

* fixed lint errors I made in tests....mqtt/test_init.py

* more lint fixes for my added test

* remove dual decode of MQTT payload

* convert if to try, except, else statement for mqtt payload decode

* rework mqtt unicode testing code to properly check for log file entriy on unicode decode exception

* fixed lint error

* Update test_init.py

kirichkov committed:
Relaxes the configuration options for influxdb (#3869)

* Relaxes the configuration options for influxdb

By default influxdb allows unauthenticated access
Home Assistant required at least username and password to be present to properly submit data to influxdb

* Removes unused import of 'copy'

The copy module was used only in the removed test case responsible for testing the missing keys

* Updates InfluxDB config schema to require user and password

Current InfluxDB (v 1.0) can work without any authentication, but when authentication is enabled both username and password should be set.

* Removes extra white space in test_influxdb.py

emlove committed:
Zwave rgb fix (#3879)

* _zw098 must be set before update_properties

* Fix problematic zwave color light value matching

See https://community.home-assistant.io/t/color-issues-with-aeotec-zwave-zw098/2830

tchellomello committed:
Added the ability to Weather Underground to track severe weather alerts (#3505)

*  Added the ability to Weather Underground to track severe weather alerts

*   * Added message on the advisory attr

  * Updated tests

* * Making use of guard clause

* Checking multiple_alerts prior loop

* Using a better way to create dict

* Fixed issue to set to None only the object that failed

* Added unittest

* Split update() method to different calls with their one throttle control to minimize API calls

* Updated unittest and make sure the alert sensor will not return 'unknown' status'

* Removed update() method from state property

* Branch rebased and include Weather Underground attribution

* Update wunderground.py

fabaff committed:
Minimum/maximum/mean sensor (#3852)

* Add min/max sensor

* Update min_max.py

fabaff committed:
Update link to docs repo (#3886)

jaharkes committed:
Added tests for the template is_defined filter

fabaff committed:
Upgrade sendgrid to 3.6.0 (#3887)

fabaff committed:
Fix name allocation (#3890)

pvizeli committed:
Async EntitiesComponent (#3820)

* first version

* First draft component entities

* Change add_entities to callback from coroutine

* Fix bug add async_prepare_reload

* Group draft v1

* group async

* bugfix

* bugfix v2

* fix lint

* fix extract_entity_ids

* fix other things

* move get_component out of executor

* bugfix

* Address minor changes

* lint

* bugfix - should work now

* make group init async only

* change update handling to old stuff

* fix group handling, remove generator from init

* fix lint

* protect loop for spaming with updates

* fix lint

* update test_group

* fix

* update group handling

* fix __init__ async trouble

* move device_tracker to new layout

* lint

* fix group unittest

* Test with coroutine

* fix bug

* now it works :100:

* ups

* first part of suggestion

* add_entities to coroutine

* change group

* convert add async_add_entity to coroutine

* fix unit tests

* fix lint

* fix lint part 2

* fix wrong import delete

* change async_update_tracked_entity_ids to coroutine

* fix

* revert last change

* fix unittest entity id

* fix unittest

* fix unittest

* fix unittest entity_component

* fix group

* fix group_test

* try part 2 to fix test_group

* fix all entity_component

* rename _process_config

* Change Group to init with factory

* fix lint

* fix lint

* fix callback

* Tweak entity component and group

* More fixes

* Final fixes

* No longer needed blocks

* Address @bbangert comments

* Add test for group.stop

* More callbacks for automation

emlove committed:
Zwave alt delay workaround for HS-WD100+ (#3893)

turbokongen committed:
ZWave: Add association service (#3894)

* Add association service

* Refactor service

* Requested changes

* Grammar in pydocstyle

balloob committed:
Add util.async to the dev docs

balloob committed:
Document more core pieces

pvizeli committed:
Async speedup add_device callback (#3910)

* Speed up entities processing from add_device callback

* fix lint

* fix bug

justweb1 committed:
Move micromarkdown to HA (#3908)

* Move micromarkdown to HA

* Fix requests

* Update micromarkdown-js.html& .gz

* Update micromarkdown-js files

persandstrom committed:
Tellstick switch force update (#3874)

fabaff committed:
Add web scrape sensor (#3841)

* Add web scrape sensor

* Add support for 'value_template', set 'verify_ssl' to true,
and remove 'before', 'after' & 'element'

* Fix pylint issue

lwis committed:
Resolve issue with delay not passing variables to render (#3901)

capellini committed:
Removing calls to mock.assert_called_once_with (#3896)

If a mock's assert_called_once_with method is misspelled (e.g. asert_called_once_with) then the test will appear as passing.  Therefore, this commit removes all instances of assert_called_once_with calls and replaces them with two assertions:

        self.assertEqual(mock.call_count, 1)
        self.assertEqual(mock.call_args, mock.call(call_args))

justweb1 committed:
Fix Comment (#3913)

joyrider3774 committed:
Emoncms history component (#3531)

* Emoncms_history component, fix git mess

* - switch to track_point_in_time to send all data at foxed interval
- don't use json_dump
- switch to http post instead of http get

pvizeli committed:
Async template (#3909)

* port binary_sensor/template

* port sensor/template

* port switch/template

* fix unittest

* fix

* use task instead yield on it

* fix unittest

* fix unittest v2

* fix invalid config

* fix lint

* fix unuset import

sdague committed:
Suppress requests/urllib3 connection pool log messages (#3854)

requests/urllib3 is notorious for using the INFO log level for very
DEBUG kinds of information. Given the configurability of python
logging it's actually pretty easy to just set requests to WARN by
default. This cleans out a bunch of largely unuseful log lines from
home assistant output.

balloob committed:
Clean up some async stuff (#3915)

* Clean up some async stuff

* Adjust comments

* Pass hass instance to eventbus

sam-io committed:
Support for Apple Push Notification Service (#3756)

* added push notification implementation

* some lint changes

* added docs

* added push notification implementation

* some lint changes

* added docs

* Fixed comment formatting issues

* Added requirments

* Update requirements_all.txt

* Update apns.py

* re-generated requirments_all.txt

* Added link to online docs

* added push notification implementation

* some lint changes

* added docs

* added push notification implementation

* some lint changes

* added docs

* Fixed comment formatting issues

* Added requirments

* Update requirements_all.txt

* Update apns.py

* re-generated requirments_all.txt

* Added link to online docs

* changed to use http/2 library for push notifications

* fixed lint issue

* fixed test that fails on CI

* another go at fixing test that fails on CI

* another go at fixing test that fails on CI

* another go at fixing test that fails on CI

* added missing docstring

* moved service description to main services.yaml file

* renamed apns service

Commit:
Protocol is an int (#3928)

tchellomello committed:
#3829 - Fixed issued LowHighTuple doesn't define __round__ method for Nest Sensor (#3881)

* #3829 - Fixed type LowHighTuple doesn't define __round__ method issue when Nest is operating in range mode

* Testing if temperature is a tuple instead int or float

gieljnssns committed:
Netatmo thermostat (#3888)

* Added Netatmo-thermostat

* Remove-CONF_DEVICES

JasonCarter80 committed:
Concord232 alarm panel (#3842)

* Adding Concord232 Alarm Panel

* Adding Concord232 Zones as Sensors

* Updating requirements_all.txt

* Adding DOCType and making helper function a closure for clarity

* Fixing D400 error in build

* Fixing pylint errors

* Adding # pylint: disable=too-many-locals

* Updating with proper polling methods

* Fixing Merge issues

* Fixing DocStyle Issue

* Moving Import out of setup

* Fixing DocString

* Removing overthought GLOBAL

juggie committed:
Ecobee - Celsius (#3906)

* #3899 - Ecobee tempoeratures

* #3899 Remove unused import

* #3899 Implement min/max_temp in ecobee.py as these temperatures have to be in F for ecobee api.

* #3899 Stale print

* #3899 Use min/max_temp from base class

* #3899 Removed unused import (again) so tests pass since changing to use super class

* #3899 Fix long lines

* #3899 Install tox locally... make it happy, commit

* #3899 Remove overridden min/max_temp and instead update __init__:min/max_temp to convert to self.temperature_unit (of the thermostat) as opposed to self.unit_of_measurement (of the system), which is wrong

* Remove unused import from ecobee

fabaff committed:
Upgrade python-digitalocean to 1.10.0 (#3921)

capellini committed:
Replacing tempfile with mock_open in tests (#3753)

- test_bootstrap.py
- test_config.py
- components/test_init.py
- components/test_panel_custom.py
- components/test_api.py
- components/notify/test_file.py
- components/notify/test_demo.py
- components/camera/test_local_file.py
- helpers/test_config_validation.py
- util/test_package.py
- util/test_yaml.py

No changes needed in:
- components/cover/test_command_line.py
- components/switch/test_command_line.py
- components/rollershutter/test_command_line.py
- components/test_shell_command.py
- components/notify/test_command_line.py

Misc changes in:
- components/mqtt/test_server.py

Also, removed some unused mock parameters in tests/components/mqtt/test_server.py.

pvizeli committed:
Bugfix Template sensors (#3931)

lwis committed:
Reset Bravia playing info to ensure state reflects correctly (#3903)

balloob committed:
Build frontend

mweinelt committed:
Add support for matrix notifications (#3827)

DavidLP committed:
New pilight sensor component (#3822)

* Pilight daemon expects JSON serializable data. Thus dict is needed and not a mapping proxy.
* Add explanation why dict as message data is needed
* Use pytest-caplog and no unittest.TestCase

pvizeli committed:
Rename add_devices to async_add_devices like dev guide (#3938)

justweb1 committed:
Fix Typo (#3942)

sdague committed:
Yamaha zones (#3920)

* Enhance yamaha component

This enhances the yamaha component to create 1 player per zone,
instead of only one play, which provides direct control of the various
zones for the player.

It also exposes play_media for NET_RADIO sources, which allows direct
setting of that.

This requires code changes in rxv 0.2.0, so the requirement dependency
is raised.

* Support current playback metadata for NET_RADIO

When on NET RADIO, support the currently playing information.

kellerza committed:
Fixup device_tracekt.mqtt voluptuous & unit tests (#3904)

emlove committed:
Logbook filtering of automations by entity_id (#3927)

* Logbook filtering of automations by entity_id

* Trigger action function parameters required

Khabi committed:
Fixing verbage (#3940)

lukas-hetzenecker commented:
**Breaking change in Z-Wave**

We unfortunately had to change the way entitiy ids are generated again with this release. For now on, they will also include the index number of the value in the entity_id. This was necessary, because before different values could have been mapped to the same id, which could have resulted in missing entities and other weird behavior of the Z-Wave component (for example, see #3509).

DavidLP committed:
Bugfixes for pilight hub component and unit tests (#3948)

lwis commented:
#3783 is a potentially breaking change for some users as config files they may not expect to be loaded will now be loaded, it's worth flagging this as a cautionary - majority of users are unlikely to be affected.

hcooper committed:
Add support for multiple inputs to nmap tracking module as a list (#3944)

pvizeli committed:
Bugfix sonos (#3926)

* Bugfix Sonos

* lint

* Use player uid for looking of exists

* fix lint

* fix unittest

* Change player_id to unique_id

fabaff committed:
Fix PEP257 issues (#3962)

henryk committed:
Fritzbox - Only report a MAC address if it's really there (#3964)

Fixes 'Neither mac or device id passed in'

tchellomello committed:
Added unit test to the Yahoo Finance sensor (#3943)

MartinHjelmare committed:
Upgrade pymysensors to 0.8 (#3954)

kellerza committed:
Updater component with basic system reporting (#3781)

JasonCarter80 committed:
Moving updates out of state property (#3966)

texnofobix committed:
Adding security contexts to the resources. (#3840)

chennin committed:
Upgrade SoCo to 0.12 (#3951)

Changelog: https://github.com/SoCo/SoCo/releases/tag/v0.12

Backwards Compatability changes:

    Dropped support for Python 3.2
    Methods relating to the music library (get_artists, get_album_artists, get_albums and others) have been moved to the music_library module. Instead of device.get_album_artists(), please now use device.music_library.get_album_artists() etc. Old code will continue to work for the moment, but will raise deprecation warnings
    Made a hard deprecation of the Spotify plugin since the API it relied on has been deprecated and it therefore no longer worked
    Dropped pylint checks for Python 2.6

T3m3z committed:
Device tracker: SNMPv3 (#3961)

* Add initial SNMPv3 support for better security

* Fixed indentation errors

* Fixed flake8 E128 on row 65

* Disabled warning about too many instance-attributes

* Removed extra code, added Inclusive to make better config validation

Commit:
Telstra SMS API notification component (#3949)

* Telstra API component

* import PLATFORM_SCHEMA

* Exclude Telstra notify component in coveragerc

* fix authentication issues

* Include title in SMS if it's provided

* pass lint

* Fix many code styling issues

* Confirm credentials are correct on component setup

turbokongen committed:
Add support for verisure file camera. (#3952)

kellerza commented on homeassistant/components/camera/verisure.py:
```diff
@@ -0,0 +1,102 @@
+"""
+Camera that loads a picture from a local file.
```
Copy/paste comment?

MartinHjelmare committed:
Fix test using libsodium and SECRET_KEY (#3975)

* Move test to class with custom config setups and with config
  validation.

kellerza committed:
Updater component - rename opt_out to reporting (#3979)

* Updater component - rename opt_out to reporting

* Fix tests

balloob committed:
Update frontend

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

fabaff committed:
Migrate to voluptuous (#3748)

jabesq committed:
Add support for Neato Connected robot as a switch (#3935)

* Add support for Neato Connected robot as a switch

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

* Add checklist items

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

* Add missing docstring

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

* [Neato] Add update function to retrieve robot state

* Add docstring for update function + catch exception when retrieving state

* Change type of HTTPError when updating state

* Fix pylint errors

HydrelioxGitHub committed:
Add Bbox Router bandwidth as sensors (#3956)

* Add Bbox Routeur bandwidth as sensors

Add possibility to monitor max and currently used bandwidth of your xdsl connection for Bbox Routeur

* Minor Fixes

Unit constant get back into the main sensor file

* Unused round removed

kellerza committed:
Improve known_device.yaml writing (#3955)

* Better known_device.yaml writing

* yaml dump

kirichkov committed:
Adds energy monitoring capabilities to the TP-Link HS110 (#3917)

* Adds energy monitoring capabilities to the TP-Link HS110

Energy monitoring works only on the HS110 model

* Reverts to using GadgetReactor's module

* Updates requirements_all.txt

* Refactors tplink switch to use attribute caching

* Update tplink.py

dasos committed:
Work better with password-protected Squeezebox / LMS servers (#3953)

* Work better with password-protected Squeezebox / LMS servers, including getting file art. Refactored to only have a single method calling the telent lib. (Should make it easier to convert to the more appropriate JSON interface)

* Update squeezebox.py

* Update squeezebox.py

* Update squeezebox.py

* Update squeezebox.py

* Update squeezebox.py

Commit:
Fixed statsd stopping if state is not numeric or only attributes changed (#3981)

* Fixed statsd stopping if attribute changed by not the state

* Fixed tests which exposed a new bug.

* Fixed another issue. whoops

robbiet480 committed:
HA iOS support (#3752)

* Initial commit of the iOS component and platform

* Allow extra

* Add battery to identify, a new function to get devices, and load the upcoming sensor

* Add iOS sensor platform, currently for battery state & level

* Add discoverability for the iOS app

* Convert single quote to double quotes

* Load all required components and platforms when loading the iOS component for the best experience

* Unify quote style to double

* Change to hass_ios

* Update push URL, add support for logging based on status code, log rate limit updates

* Block iOS from coverage checks for now...

balloob committed:
Fix updater, add new fields (#3982)

* Fix updater

* Add Docker and virtualenv checks

* Add log line informing user of update/analytics

* Remove str

ehagan committed:
Adds support for Pioneer AVR interface port number (#3878)

* Adds support for Pioneer AVR interface port number

https://community.home-assistant.io/t/support-for-pioneer-avr/503
telnetlib supports a port number so adding port as
an optional config element with a default of 23 resolves this.

* Adds timeout to Pioneer AVR

timeout in telnetlib defaults to socket._GLOBAL_DEFAULT_TIMEOUT
which is not a value, but rather a bare Object used for comparison.

telnetlib says the following about the timeout optional argument:
"The optional timeout parameter specifies a timeout in seconds
 for blocking operations like the connection attempt (if not
 specified, the global default timeout setting will be used)."

From the documentation for sockets:
"Sockets are by default always created in blocking mode"

Catching connect and timeout errors, logging to debug
and continuing.

* Catches timeout exceptions, logs and continues.

DavidLP commented:
When we go down the road of user data collection I would recommend also to setup some CI for bug collection like [bugsnag](https://bugsnag.com/), [rollbar](https://rollbar.com/), or [sentry](https://sentry.io).
This would automatically and very quickly show the most common bugs. Unfortunately these service are not available for free for the large home assistant user base.

lukas-hetzenecker commented:
> When we go down the road of user data collection we might also setup some CI for bug collection like bugsnag, rollbar, or sentry.

I think collecting specific installation data (like OS release, architecture) is okay, but I wouldn't be comfortable if possibly highly sensitive data from bugs would be sent to some sentry server automatically. 

As an alternative, I'd suggest to show a persistent notification if a component fails, with a link to open a bug (if possible, even with a filled out bug reporting template).

turbokongen committed:
Prevent zwave from firing event at shutdown (#3987)

jabesq commented:
Just a small edit for the release note, it should be "Netatmo Welcome camera: add binary sensor" instead of "Netatmo weather station: add binary sensor" as the datas come from from the Welcome camera and not the weather station

fabaff commented:
@jabesq thanks

mweinelt commented:
Change
```
HTTP: Change `approved_ipds` from string to CIDR validation (@mweinelt)
```
to
```
HTTP: Change `approved_ips` from string to CIDR validation (@mweinelt)
```

balloob committed:
Update frontend

turbokongen committed:
Prevent Verisure cam to delete a file when it is None (#3988)

Commit:
Update wink.py (#3957)

* Update wink.py

added lambda and smoke detector call in component loading routine to fix broken functionality.

* Update wink.py

fixed extra space.

* Update wink.py

applied cleaner refactor per comments

* Update wink.py

fixed spacing

* Update wink.py

fixed lint error #1

robbiet480 committed:
Update version
